equational_theories/Generated/All4x4Tables/src/generate_lean.py

```diff
@@ -146,22 +159,47 @@ def «{name}» : Magma (Fin {div}) where
 """
     return out
 
-
-with open(f"{dir}/data/refutations.txt") as f:
-    with open(f"{dir.parent}/All4x4Tables.lean", "w") as main:
-      lines = f.readlines()
-      # the format is groups-of-three-lines-based
-      lines = [lines[i:i + 3] for i in range(0, len(lines), 3)]
-      for i, line in enumerate(lines):
-          leanfile = f"{dir}/Refutation{i}.lean"
-          data = parse_row(line)
-          if data and data["div"] < 6:
-            data = prune_row(data)
-            print(f"Writing {leanfile}")
-            main.write(f"import equational_theories.Generated.All4x4Tables.Refutation{i}\n")
-            with open(leanfile, "w") as f:
-                  f.write(generate_lean(data))
-
+def create_rows(f):
+    first_line = f.readline().strip()
+    all_rows = []
+    row = []
+    if not first_line.startswith("Plan"):
+        # processing a dump_tables output file
+        print("Reading dump_tables output...")
+        if first_line.startswith("Table") or first_line.startswith("Proves"):
+            row.append(first_line)
+        for line in f:
+            line = line.strip()
+            if line.startswith("Table") or line.startswith("Proves"):
+                row.append(line)
+            if len(row) == 2:
+                all_rows.append(row)
+                row = []
+    else:
+        # processing a make-plan output file
+        print("Reading make-plan output...")
+        for line in f:
+            line = line.strip()
+            if line.startswith("Magma") or line.startswith("Satisfies") or line.startswith("Refutes"):
+                row.append(line)
+            if len(row) == 3:
+                all_rows.append(row)
+                row = []
+    return all_rows
+
+rows = []
+with open(f"{dir}/data/plan.txt") as f:
+    rows = create_rows(f)
```

Comment:
No need to use `with` if its only used once anyways:
```suggestion
rows = create_rows(open(f"{dir}/data/plan.txt"))
```